sub-packages/bionemo-llm/src/bionemo/llm/utils/callbacks.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+from typing import Any, Literal, Sequence
+
+import pytorch_lightning as pl
+import torch
+from lightning.pytorch.callbacks import BasePredictionWriter
+
+from bionemo.llm.lightning import batch_collator
+
+
+IntervalT = Literal["epoch", "batch"]
+
+
+class PredictionWriter(BasePredictionWriter, pl.Callback):
```

Comment:
Can you add tests for this somewhere? Ideally we'd use this in both the DDP and single-device paths, otherwise it's going to be hard to write tests for this that run in CI

Reply:
yes, makes sense, added here: [6a18c19](https://github.com/NVIDIA/bionemo-framework/pull/482/commits/6a18c19d8bbaf98c3612b450024584de099d779f)